base/dict.jl

```diff
@@ -189,6 +189,16 @@ function grow_to!{K,V}(dest::Associative{K,V}, itr, st)
     return dest
 end
 
+
+"""
+    similar(d::Dict, [element_type=Pair{K,V}])
+
+Create an empty dictionary with key type `K` and value type `V`. The second element
```

Comment:
I even have an hard time to come up with a meaningful example. Read the following:
```
    similar(d::Dict, element_type=eltype(d))
 
Create an empty dictionary with the same key and value types as `d`.
# Example
Create an empty dictionary `Dict{Int,Float64}()` by passing the element type `Pair{Int,Float64}`:
julia> similar(d, Pair{Int,Float64})
Dict{Int,Float64}()
```
to which anybody would say: why not `Dict{Int,Float64}()` and we are done? Without the note about similar and the generic code, the method seems useless. Do you have a better suggestion?